src/lib.rs

```diff
@@ -122,3 +288,23 @@ impl Config {
         self
     }
 }
+
+/// Unfortunately due to the API limitations of `RangeBound` we can only ever get an `&` (not a
+/// `&mut` or owned value) of a `Bound<T>`. So in order to coherce the value into the bound we want
+/// we must clone the data.
+/// 
+/// The only way to avoid this would be to only accept two `Bound<T>` arguments, not one
+/// `RangeBound<T>` argument. This would mean `"abc"..="xyz"` would not be possible.
+/// 
+// TODO: If you feel tricky please try to remove this clone!
```

Comment:
PTAL @sunxiaoguang

Reply:
Yea, I think this way is better it can avoid extra clone when bound is already a Key however make it possible to use non Key type which can be converted to Key.

Reply:
Great idea, I'll integrate this @sticnarf